Ashwalker Tribe roundend reporting revamp

Rhials opened the pull request:
## About The Pull Request

This PR changes how ashwalkers are listed in the roundend report. Originally, they would be listed as individual "other" antagonists, which isn't a particularly great way of handling it and leads to weirdness like this:

![image](https://github.com/tgstation/tgstation/assets/28870487/c00a2cfc-3f08-46f6-a290-ef632cac0371)

_(That's an authentic ninja greentext from a live server, by the by)_

Now, the roundend report gives each ashwalker team their own section within the report, whether or not they succeeded, and some bonus stats.

![image](https://github.com/tgstation/tgstation/assets/28870487/c2dc3c00-62a7-4784-90be-5a8bb5cc5a50)

This new roundend report handling also supports having multiple Necropolises (Necropoli?) in the same round. Each tribe has their stats and success reported individually. 

This also slightly modifies the protect_object objective, preventing hard deletes when linked to objects that don't clear their own references.

## Why It's Good For The Game

Gives Ashwalkers a much better roundend report, and handles how the reporting is done in a much more sensible manner.

Reduces how much space that large nests take up with their individual listings.
## Changelog
:cl: Rhials
qol: Ashwalker roundend reporting has been revamped. Glory to the Necropolis!
/:cl:

Rhials committed:
first

Rhials committed:
finishes that

Rhials committed:
makes the messages beetter

adds something for when nobody takes an egg too

Rhials committed:
stattrak lavaland tendril poggies

Rhials committed:
stop getting rid of that

Rhials committed:
apostrophe

Rhials committed:
I probably didnt need to change this in retrospect

Rhials committed:
forgot about the actual part where it gets deleted

Rhials committed:
nulls a reference I missed

Helg2 commented:
Maybe add namings for tribes? So it's also easier to find the tribe you're looking for.

Rhials committed:
okay so

since I reworked how the ashwalker tendril handles the ashwalker objective (originally it deleted the objective when the tendril was destroyed).

I removed this behavior because I wanted the objective to persist for roundend reporting. It appears that the protect_object objective would always hard delete unless the object itself cleared the objective before deleting itself.

So, hopefully, this should work.

Googles-Hands commented:
Why change it from Ash Walker to Ashwalker. The imdividual antags are still called "Ash Walker", either change it all or keep it uniform.

Rhials committed:
a wonderful assessment, mister hands

san7890 commented on code/modules/antagonists/ashwalker/ashwalker.dm:
```diff
@@ -42,3 +38,40 @@
+/datum/team/ashwalkers/roundend_report()
+	var/list/report = list()
+
+	report += "<span class='header'>An Ash Walker Tribe inhabited the wastes...</span><br>"
```
do you think it'd be worthwhile to make a `span_header()` for this in this PR? it should definitely be done at some point so i'm wondering if now is a good time or not.

Mothblocks replied:
im killing this whole menu when i get the motivation anyway

san7890 commented on code/modules/antagonists/ashwalker/ashwalker.dm:
```diff
@@ -42,3 +38,40 @@
+
+		if(necropolis_objective)
+			objectives -= necropolis_objective //So we don't count it in the check for other objectives.
+			report += "<b>The [name] was tasked with defending the Necropolis:</b>"
```
these could reasonably be `span_bold()`, pretty sure our HTML ships with class=bold (if it doesn't: wack).

Rhials replied:
I tried using span_bold(), it didn't seem to do anything.

I concur, this is indeed "wack".

Rhials committed:
adds a span header macro

AviiCinders commented:
The tribe has fallen, billions must die
